semgrep/java/java-use-of-rsa-encryption-algorithm-with-short-key.yml

```diff
@@ -0,0 +1,22 @@
+- id: java-use-of-rsa-encryption-algorithm-with-short-key
+  message: |
+    It is not secure to use the RSA algorithm with a 1024-bit key. The minimal acceptable key length is 2048 bits.
+  languages: [ java ]
+  severity: ERROR
+  metadata:
+    vulnerability_category: weak_cryptography
+    violation_id: weak_encryption_algorithm
+    standards: [ Microsoft SDL ]
+    references:
+      - https://docs.microsoft.com/en-us/security/sdl/cryptographic-recommendations
+  patterns:
+    - pattern: |
+        $K = $C.getInstance("$RE");
```

Comment:
can we change this to something like:
- pattern-inside: |
    $K = $C.getInstance("$RE");
    ...
- pattern: $K.initialize($BITS);
- fix: $K.initialize(2048);

and this way we can have an auto-fix as well?